kasa/smart/modules/cleanrecords.py

```diff
@@ -0,0 +1,184 @@
+        metadata=field_options(lambda x: timedelta(minutes=x))
+    )
+    total_area: int
+    total_count: int = field(metadata=field_options(alias="total_number"))
+
+    records: list[Record] = field(metadata=field_options(alias="record_list"))
+    last_clean: Record = field(
+        metadata=field_options(
+            serialization_strategy=LastCleanStrategy(), alias="lastest_day_record"
+        )
+    )
+
+
+class CleanRecords(SmartModule):
+    """Implementation of vacuum cleaning records."""
+
+    REQUIRED_COMPONENT = "clean_percent"
+    _parsed_data: Records
+
+    async def _post_update_hook(self) -> None:
+        """Cache parsed data after an update."""
+        self._parsed_data = Records.from_dict(self.data)
+
+    def _initialize_features(self) -> None:
+        """Initialize features."""
+        for type_ in ["total", "last"]:
+            self._add_feature(
+                Feature(
+                    self._device,
+                    id=f"{type_}_clean_area",
+                    name=f"{type_.capitalize()} area cleaned",
+                    container=self,
+                    attribute_getter=f"{type_}_clean_area",
+                    unit_getter="area_unit",
+                    category=Feature.Category.Debug,
+                    type=Feature.Type.Sensor,
+                )
+            )
+            self._add_feature(
+                Feature(
+                    self._device,
+                    id=f"{type_}_clean_time",
+                    name=f"{type_.capitalize()} time cleaned",
+                    container=self,
+                    attribute_getter=f"{type_}_clean_time",
+                    category=Feature.Category.Debug,
+                    type=Feature.Type.Sensor,
+                )
+            )
+        self._add_feature(
+            Feature(
+                self._device,
+                id="total_clean_count",
+                name="Total clean count",
+                container=self,
+                attribute_getter="total_clean_count",
+                category=Feature.Category.Debug,
+                type=Feature.Type.Sensor,
+            )
+        )
+        self._add_feature(
+            Feature(
+                self._device,
+                id="last_clean_timestamp",
+                name="Last clean timestamp",
+                container=self,
+                attribute_getter="last_clean_timestamp",
+                category=Feature.Category.Debug,
+                type=Feature.Type.Sensor,
+            )
+        )
+
+    def query(self) -> dict:
+        """Query to execute during the update cycle."""
+        return {
+            "getCleanRecords": {},
+        }
+
+    @property
+    def total_clean_area(self) -> Annotated[int, FeatureAttribute()]:
+        """Return total cleaning area."""
+        return self._parsed_data.total_area
+
+    @property
+    def total_clean_time(self) -> timedelta:
+        """Return total cleaning time."""
```

Comment:
Unsure what's the best approach for homeassistant support. It's a duration, but we need to convert this somehow into minutes for consumption.